reference-apps/aaos-unbundled/mediacompose/src/main/java/com/android/designcompose/reference/mediacompose/MediaAdapter.kt

```diff
@@ -415,6 +420,10 @@ class MediaAdapter(
 
         val currentlyPlaying =
             if (isMetadataSame(item, currentMetadata)) CurrentlyPlaying.On else CurrentlyPlaying.Off
+
+        // Unsubscribe this setter function whenever this leaves the composition
+        DisposableEffect(setIcon) { onDispose { artRequestManager.unsubscribe(setIcon) } }
```

Comment:
I think ideally the effect would also perform the subscription (or the subscribe-unsubscribe thing would be encapsulated in a composable function) just so they can't get out of sync. I never remember if `DisposableEffect` executes eagerly, or if it runs after `MediaBrowseItem` finished...).

Reply:
This was a bit messy due to the removal of `@Composable`. I subscribe in the image replacement context function because that's when I know the size and color to make the image that I want. I don't have that info here. However to ensure we don't leak memory and have extra subscriptions when this composable leaves composition, I added this `DisposableEffect`.

Reply:
Resolving so I can get this change in. If you think this needs to be changed, we can file another ticket to make the changed, but I'm not sure how to move the subscription into the @Composable function.